docs/docs/javascript/overview/eslint.mdx

```diff
@@ -0,0 +1,554 @@
+]
+```
+
+### TypeScript Support
+
+```toml title="pants.toml"
+[GLOBAL]
+backend_packages.add = [
+  "pants.backend.javascript",  # Required for JavaScript support
+  "pants.backend.typescript",  # Required for TypeScript support
+  "pants.backend.experimental.javascript.lint.eslint",  # ESLint integration
+]
+```
+
+
+## Configuration
+
+### Basic Setup
+
+ESLint will automatically discover configuration files following the standard ESLint resolution order:
+
+1. Flat config files (ESLint 9.0+): `eslint.config.js`, `eslint.config.mjs`, `eslint.config.cjs`
+2. Legacy config files: `.eslintrc`, `.eslintrc.js`, `.eslintrc.cjs`, `.eslintrc.yaml`, `.eslintrc.yml`, `.eslintrc.json`
+3. `eslintConfig` section in `package.json`
+
+### Options
+
+ESLint behavior can be configured through `pants.toml`:
+
+```toml title="pants.toml"
+[eslint]
+# ESLint version to use
+version = "eslint@8.57.0"
+
+# Additional arguments to pass to ESLint
+args = ["--max-warnings=0", "--cache"]
+
+# Explicit config file path (disables automatic discovery)
+config = "custom-eslint.config.js"
+
+# Enable/disable automatic config discovery
+config_discovery = true
+
+# Skip linting and formatting entirely
+skip = false
+
+# Enable JSON output with summary parsing
+json_output = true
+
+# Additional TypeScript/plugin dependencies
+extra_dev_dependencies = [
+    "@typescript-eslint/parser@^6.0.0",
+    "@typescript-eslint/eslint-plugin@^6.0.0"
+]
+```
+
+### Version and Requirements
+
+By default, Pants uses ESLint version 8.57.0. You can customize this:
+
+```toml title="pants.toml"
+[eslint]
+version = "eslint@9.0.0"
+```
+
+## Usage
+
+### Linting
+
+Run ESLint to check for code quality issues:
+
+```bash
+$ pants lint src/js/::
+```
+
+### Formatting
+
+Run ESLint with `--fix` to automatically fix formatting issues:
+
+```bash
+$ pants fmt src/js/::
+```
+
+## BUILD File Configuration
+
+### JavaScript Sources
+
+For JavaScript files, use the `js_sources` target:
+
+```python title="src/js/BUILD"
+js_sources(
+    name="frontend",
+    sources=["**/*.js", "**/*.mjs", "**/*.cjs"],
+)
+```
+
+### TypeScript Sources
+
+For TypeScript files, use the `typescript_sources` target:
+
+```python title="src/ts/BUILD"
+typescript_sources(
+    name="backend",
+    sources=["**/*.ts"],
+)
+```
+
+
+### Mixed Projects
+
+For projects with multiple file types:
+
+```python title="src/BUILD"
+js_sources(
+    name="js_files",
+    sources=["**/*.js"],
+)
+
+typescript_sources(
+    name="ts_files",
+    sources=["**/*.ts"],
+)
+```
+
+### Target-specific Configuration
+
+You can skip ESLint for specific targets using the `skip_eslint` field:
+
+```python title="src/js/BUILD"
+js_sources(
+    name="legacy",
+    sources=["legacy/*.js"],
+    skip_eslint=True,  # Skip linting for legacy code
+)
+
+typescript_sources(
+    name="modern",
+    sources=["modern/*.ts"],
+    # ESLint will run on these files
+)
+```
+
+## Configuration Files
```

Comment:
It might be better here to link to the eslint docs on config file formats and just state that Pants supports them (to avoid re-documenting here, plus this doc is already quite long).